### PR TITLE
 implemented modules override mechanism

### PR DESCRIPTION
**Pull request by rwngwn**

### CODE REVIEW

File: cct/module.py [hunk 1]
```diff
@@ -97,21 +104,53 @@ def check_module_py(self, candidate):
...
+            logger.error("Conflicting module '%s' version found, installed version '%s', required version '%s'" % (name, self.modules[name].version, self.version))
+            raise Exception('Conflicting modules')
+        return True
 
     def list(self):
         print("available cct modules:")
-        for module, _ in self.modules.iteritems():
-            print("  %s" % module)
+        for name, module in self.modules.iteritems():
+            if name == 'base.Dummy':
```
Comment (goldmann) [inline cct/module.py:147]: What's up with this?

Comment (rwngwn) [reply cct/module.py:147]: dummy is special module used a lot in test - It needs to be moved to better place in future

Comment (goldmann) [reply cct/module.py:147]: Yes, it needs :)

File: cct/module.py [hunk 2]
```diff
@@ -97,21 +104,53 @@ def check_module_py(self, candidate):
...
+        if name not in self.modules:
+            return False
+        # if module was froced we will not override it
+        if self.modules[name].override:
+            return True
+        # check if the version are not conflicting
+        # FIXME - tag in branch vs branch is considered incompatible
+        if self.modules[name].version != self.version:
+            logger.error("Conflicting module '%s' version found, installed version '%s', required version '%s'" % (name, self.modules[name].version, self.version))
+            raise Exception('Conflicting modules')
```
Comment (goldmann) [inline cct/module.py]: Why not raise above message?

Comment (rwngwn) [reply cct/module.py]: good catch will update it

**Pull request merged**